### DISCUSSION TITLE
Difference in "need to specify version" between jsr and deno.land

### DISCUSSION THREAD

**Discussion by Xernisfy** [denoland/deno#27445] [General]
Hi Deno community,

since the launch of JSR 🎉 I've noticed, that you don't have to (or even should?) specify a version when importing a dependency.
You just `import { dirname } from 'jsr:@std/path';` and it will resolve to the latest version by default.
When compared to `deno.land` it's _recommended_ that you specify a version.
Otherwise you get squiggly lines on the import:
![image](https://github.com/user-attachments/assets/d766fe55-c107-4b9e-9142-2d8ca475a135)

The resolution works completely fine either way:
|Service|Specific version|No version|
|-|-|-|
|jsr|works ✔️|works ✔️|
|deno.land|works ✔️|works ✔️|

I quite like not having to specify a version, as it checks for the newest dependencies on each run (if you don't use a lockfile).

But why is there a difference in how versions are expected?
Is the warning for "Implicitly using latest version..." necessary?

Kind regards

**Comment by marvinhagemeister** [comment]
You can skip adding a version if:
- you're writing a simple app or one of script
- you are prototyping something
- don't care if your app keeps working in the future (because it's a prototype etc)

You should _always_ use a version or version constraint together with a lockfile if:

- more than one dev work on the code
- the app should keep working in the future
- you don't want to spent time debugging why the app works on machine A, but not on machine B (this typically happens because of different dependency version installed on the users machine. A lockfile prevents this)

We strongly recommend going with the versioned approach for any serious project. Only consider going without versions if it's a short lived prototype that only a single developer hacks on.

**Reply by karfau** [reply]
Maybe worth mentioning in that context (because it's hard to find in the docs),
that it's possible to specify a version AND a path into the package, e.g.

```js
export { describe, it } from "jsr:@std/testing@1.0.16/bdd";
```
